src/Rules/Preg/RegexPatternRule.php

```diff
@@ -0,0 +1,119 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Preg;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Type\Constant\ConstantArrayType;
+use PHPStan\Type\Constant\ConstantStringType;
+
+class RegexPatternRule implements \PHPStan\Rules\Rule
+{
+
+	public function getNodeType(): string
+	{
+		return FuncCall::class;
+	}
+
+	/**
+	 * @param FuncCall $node
+	 * @param \PHPStan\Analyser\Scope $scope
+	 * @return string[]
+	 */
+	public function processNode(Node $node, Scope $scope): array
+	{
+		$patterns = $this->extractPatterns($node, $scope);
+
+		$errors = [];
+		foreach ($patterns as $pattern) {
+			if ($this->validatePattern($pattern)) {
+				continue;
+			}
+
+			$errors[] = sprintf('The regex pattern \'%s\' is not valid.', $pattern);
+		}
+
+		return $errors;
+	}
+
+	/**
+	 * @param FuncCall $functionCall
+	 * @param Scope $scope
+	 * @return string[]
+	 */
+	private function extractPatterns(FuncCall $functionCall, Scope $scope): array
+	{
+		if (!isset($functionCall->args[0])) {
+			return [];
+		}
+		$patternNode = $functionCall->args[0]->value;
+		$patternType = $scope->getType($patternNode);
+
+		if (!$functionCall->name instanceof Node\Name) {
+			return [];
+		}
+		$functionName = (string) $functionCall->name;
+
+		if (
+			$patternType instanceof ConstantStringType
+			&& in_array($functionName, [
+				'preg_match',
+				'preg_match_all',
+				'preg_split',
+				'preg_grep',
+				'preg_replace',
+				'preg_replace_callback',
+				'preg_filter',
+			], true)
+		) {
+			return [$patternType->getValue()];
+		}
+
+		if (
+			$patternType instanceof ConstantArrayType
+			&& in_array($functionName, [
+				'preg_replace',
+				'preg_replace_callback',
+				'preg_filter',
+			], true)
+		) {
+			$patterns = [];
+			foreach ($patternType->getValueTypes() as $arrayKeyType) {
+				if (!$arrayKeyType instanceof ConstantStringType) {
+					continue;
+				}
+
+				$patterns[] = $arrayKeyType->getValue();
+			}
+			return $patterns;
+		}
+
+		if (
+			$patternType instanceof ConstantArrayType
+			&& $functionName === 'preg_replace_callback_array'
+		) {
+			$patterns = [];
+			foreach ($patternType->getKeyTypes() as $arrayKeyType) {
+				if (!$arrayKeyType instanceof ConstantStringType) {
+					continue;
+				}
+
+				$patterns[] = $arrayKeyType->getValue();
+			}
+			return $patterns;
+		}
+
+		return [];
+	}
+
+	private function validatePattern(string $pattern): bool
+	{
+		try {
+			return preg_match($pattern, '') !== false;
```

Comment:
I don't think `preg_match` can throw an exception. I'd rather be if you used `Nette\Utils\Strings::match`and caught only the Regexp-related exception here.

It would also be cool to add the exception message to be part of the reported error message :)

Reply:
> It would also be cool to add the exception message to be part of the reported error message :)

Looks like this will be though to test due to the error messages being different per PHP version:
https://travis-ci.org/phpstan/phpstan/jobs/449075422

Reply:
Writen a different test for each version and use `self::markTestSkipped` in them :)